eth_enr/sqlite3_db.py

```diff
@@ -0,0 +1,334 @@
+import datetime
+import logging
+import operator
+import sqlite3
+from typing import Collection, Iterable, NamedTuple, Tuple, Union
+
+from eth_typing import NodeID
+import rlp
+from rlp.exceptions import DecodingError, DeserializationError, SerializationError
+
+from eth_enr.abc import ENRAPI
+from eth_enr.enr import ENR
+from eth_enr.sedes import ENR_KEY_SEDES_MAPPING
+
+logger = logging.getLogger("eth_enr.sqlite3")
+
+RECORD_CREATE_STATEMENT = """CREATE TABLE record (
+    node_id BLOB NOT NULL,
+    sequence_number INTEGER NOT NULL,
+    signature BLOB NOT NULL,
+    created_at DATETIME NOT NULL,
+    PRIMARY KEY (node_id, sequence_number),
+    CONSTRAINT _sequence_number_positive CHECK (sequence_number >= 0)
+)
+"""
+
+RECORD_INDEXES_AND_CONSTRAINTS = (
+    "CREATE UNIQUE INDEX ix_node_id_sequence_number ON record (node_id, sequence_number)",
+)
+
+FIELD_CREATE_STATEMENT = """CREATE TABLE field (
+    node_id BLOB NOT NULL,
+    sequence_number INTEGER NOT NULL,
+    "key" BLOB NOT NULL,
+    value BLOB NOT NULL,
+    PRIMARY KEY (node_id, sequence_number, "key"),
+    CONSTRAINT uix_node_id_key UNIQUE (node_id, sequence_number, "key"),
+    FOREIGN KEY(node_id) REFERENCES record (node_id),
+    FOREIGN KEY(sequence_number) REFERENCES record (sequence_number)
+)
+"""
+
+FIELD_INDEXES_AND_CONSTRAINTS = (
+    'CREATE UNIQUE INDEX ix_node_id_sequence_number_key ON field (node_id, sequence_number, "key")',  # noqa: E501
+)
+
+
+def create_tables(conn: sqlite3.Connection) -> None:
```

Comment:
I made the somewhat implicit decision not to do versioning (and definitely not migrations between versions).

1. I don't expect the schema to change.
2. If the schema changes it will be up to the end-user to use a new database or manually migrate if the records are important enough.

Reply:
I'll document this.